api/src/groups/api.py

```diff
@@ -5,527 +5,507 @@
-        db.session.delete(member)
-    db.session.delete(meeting)
+    db.session.delete(manager)
     db.session.commit()
 
-    return jsonify(msg='Meeting ' + meeting_id + ' deleted'), 200
+    # no content should be in the response with 204 status code
+    return "Deleted successfully", 204
+
+# ---- Meeting
+
+meeting_schema = MeetingSchema()
 
 
-@groups.route('/meetings/activate/<meeting_id>', methods=['PUT'])
+@groups.route('/meetings', methods=['POST'])
 @jwt_required
-def activate_meeting(meeting_id):
-    meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
+def create_meeting():
+    try:
+        valid_meeting = meeting_schema.load(request.json, partial=['active'])
+    except ValidationError as err:
+        return jsonify(err.messages), 422
+
+    if 'active' not in valid_meeting:
+        valid_meeting['active'] = True
+
+    new_meeting = Meeting(**valid_meeting)
+
+    db.session.add(new_meeting)
+
+    try:
+        db.session.commit()
+    # when the foreign key given is invalid
+    except IntegrityError:
+        return jsonify('Payload contains an invalid foreign key'), 404
 
+    return jsonify(meeting_schema.dump(new_meeting)), 201
+
+@groups.route('/meetings')
+def read_all_meetings():
+    query = db.session.query(Meeting)
+    try:
+        meetings = get_all_queried_entities(query, request.args)
+    except QueryArgumentError as e:
+        return jsonify(e.message), e.code
+    meeting_schema = MeetingSchema()
+    return jsonify(meeting_schema.dump(meetings, many=True)), 200
+
+@groups.route('/meetings/<int:meeting_id>')
+@jwt_required
+def read_one_meeting(meeting_id):
+    meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
     if meeting is None:
-        return jsonify(msg="Meeting not found"), 404
+        return jsonify(f"Meeting with id {meeting_id} does not exist"), 404
+    return jsonify(meeting_schema.dump(meeting)), 200
 
-    setattr(meeting, 'active', True)
-    db.session.commit()
-    return jsonify(meeting_schema.dump(meeting))
+@groups.route('/meetings/<int:meeting_id>', methods=['PATCH'])
+@jwt_required
+def update_meeting(meeting_id):
+    meeting_schema = MeetingSchema()
+    try:
+        valid_attributes = meeting_schema.load(request.json, partial=True)
+    except ValidationError as err:
+        return jsonify(err.messages), 422
 
+    return modify_entity(Meeting, meeting_schema, meeting_id, valid_attributes)
 
-@groups.route('/meetings/deactivate/<meeting_id>', methods=['PUT'])
+@groups.route('/meetings/<int:meeting_id>', methods=['DELETE'])
 @jwt_required
-def deactivate_meeting(meeting_id):
+def delete_meeting(meeting_id):
     meeting = db.session.query(Meeting).filter_by(id=meeting_id).first()
 
     if meeting is None:
-        return jsonify(msg="Meeting not found"), 404
+        return jsonify(f"Meeting with id #{meeting_id} does not exist"), 404
 
-    setattr(meeting, 'active', False)
+    db.session.delete(meeting)
```

Comment:
Here's a case where in some cases, we may actually want to delete a meeting. For example, if a meeting was scheduled and never held and nobody was marked as attending the meeting, it was _probably_ created in error and could be removed. However, if the meeting took place, had attendees, etc., it should probably not be allowed to be deleted.

Reply:
Like I said in the previous comment, I'm _personally_ still leaning toward making the API endpoints as explicit as possible (e.g. a DELETE endpoint actually _delete_s the entity), and implement some "trash can" functionality in the UI, utilizing those endpoints. 
But we can totally change it to the other way, if needed!

Reply:
Let's go with your plan to be explicit about the endpoints.

Please update the RESTful documentation in the `coding.md` document to reflect this change (i.e., use `PATCH` unless you _really_ want to something to go away, in which case use `DELETE`).